pkg/segment/query/iqr/intermediateQueryResult.go

```diff
@@ -356,6 +358,10 @@ func (iqr *IQR) Append(other *IQR) error {
 		return err
 	}
 
+	iqr.AddColumnsToDelete(other.deletedColumns)
+
+	iqr.AddColumnIndex(other.columnIndex)
+
 	mergedIQR, err := mergeMetadata([]*IQR{iqr, other})
```

Comment:
This seems like it should be part of `mergeMetadata`. And do you need to do any validation between the two IQRs before you merge them (e.g., they should already have the same `deletedColumns`, or should you blindly accept the `other.deletedColumns`)?

Reply:
Yes you can accept other deleted columns, since some IQRs might have different columns based on the rrcs/segments they have.

Reply:
For example: you do an -a*, IQR_1 has `address`, IQR_2 has `application`. In these cases, you won't have same deleted columns and there is no way to validate it. The best approach is to add both of them.

Reply:
addressed in the latest commit.